packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -177,41 +177,50 @@ class RequestableSizeWidget extends StatelessWidget {
 /// Wrapper to get the name of an ObjRef depending on its type.
 String? _objectName(ObjRef? objectRef) {
   String? objectRefName;
-  if (objectRef == null) return null;
 
-  if (objectRef is ClassRef ||
-      objectRef is FuncRef ||
-      objectRef is FieldRef ||
-      objectRef is LibraryRef) {
+  if (objectRef is ClassRef || objectRef is FuncRef || objectRef is FieldRef) {
     objectRefName = (objectRef as dynamic).name;
+  } else if (objectRef is LibraryRef) {
+    objectRefName =
+        objectRef.name?.isEmpty ?? false ? objectRef.uri : objectRef.name;
+  } else if (objectRef is ScriptRef) {
+    objectRefName = fileNameFromUri(objectRef.uri);
   } else if (objectRef is InstanceRef) {
     objectRefName = objectRef.name ??
         'Instance of ${objectRef.classRef?.name ?? '<Class>'}';
+  } else {
+    objectRefName = objectRef?.vmType;
   }
 
   return objectRefName;
 }
 
-/// Recursively gets the full owner name of a field or function object.
-String? _ownerName(ObjRef? ref) {
+/// Recursively gets the name of a function, qualified with its full owner name,
+/// excluding the owner library.
+String? qualifiedName(ObjRef? ref) {
   if (ref == null) return null;
-  if (ref is FuncRef) {
-    return '${_ownerName(ref.owner)}.${_objectName(ref)}';
-  } else if (ref is FieldRef) {
-    return '${_ownerName(ref.owner)}.${_objectName(ref)}';
-  } else {
-    return _objectName(ref) ?? '<unknown>';
+
+  if (ref is ClassRef) {
+    return '${ref.name}';
+  } else if (ref is FuncRef) {
+    if (ref.owner is! LibraryRef) {
+      return '${qualifiedName(ref.owner)}.${ref.name}';
+    } else {
+      return '${ref.name}';
+    }
   }
+
+  return '';
```

Comment:
Should we ever get to this case? If we don't expect to, maybe consider throwing an exception.